feat: Implements the ability to assign customer id to cart

rostyk-kanafotskyy opened the pull request:
## Type

* ### Feature
  Implements the ability to assign customer id to cart


## Notes

`POST /v2/carts/:cart_id/relationships/customers` returns `null`
 But `nock` didn't allow that, so I used `{}` for the test: https://github.com/moltin/js-sdk/pull/346/files#diff-8452f0531ec683589e7912ada26baf3cR800

dusanradovanovic reviewed (changes requested):
Let's open this same PR against `beta` branch first so that it is first published as beta. Also make sure to edit message in one commit and use semantic-release format. You can undo one commit and then run `npx git-cz` which will guide you through the process.

rostyk-kanafotskyy commented:
> Let's open this same PR against `beta` branch first so that it is first published as beta. Also make sure to edit message in one commit and use semantic-release format. You can undo one commit and then run `npx git-cz` which will guide you through the process.

Done 
Squashed 2 commits into one with the right formatting 
Rebased against the beta branch
